pkg/kv/kvserver/rafttrace/rafttrace.go

```diff
@@ -40,12 +40,27 @@ var MaxConcurrentRaftTraces = settings.RegisterIntSetting(
 	settings.IntInRange(0, 1000),
 )
 
+// LogRaftEventsToCockroachLog controls whether we log raft traces to the
+// cockroach.log in addition to adding the event to the trace. Traces are only
+// created on the leaseholder but logging is done on both leaders and followers
+// if the setting is enabled and the trace is registered. The potential downside
+// of enabling this setting is the churn it can cause in cockroach.log.
+var LogRaftEventsToCockroachLog = settings.RegisterBoolSetting(
+	settings.SystemOnly,
+	"kv.raft.log_to_cockroach_log",
+	"when true, log raft traces to the cockroach log in addition to the trace",
+	true,
+)
+
 // traceValue represents the trace information for a single registration.
 type traceValue struct {
 	traced kvserverpb.TracedEntry
 	// ctx is a trace specific context used to log events on this trace.
 	ctx context.Context
 
+	// alsoLog is true if we should log the trace to the cockroach log.
+	alsoLog bool
```

Comment:
Is `ctx` above unused if `alsoLog == false`? If so, consider just setting it to nil instead of having the bool flag.

Reply:
Good point. I changed to use `ctx==nil` to represent that we aren't tracing and removed this bool.